wpilibc/src/main/native/cpp/shuffleboard/ShuffleboardComponentBase.cpp

```diff
@@ -0,0 +1,71 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "frc/shuffleboard/ShuffleboardComponentBase.h"
+
+using namespace frc;
+
+ShuffleboardComponentBase::ShuffleboardComponentBase(
+    ShuffleboardContainer& parent, wpi::StringRef title, wpi::StringRef type)
+    : ShuffleboardValue(title), m_parent(parent) {
+  m_type = type;
+}
+
+void ShuffleboardComponentBase::SetType(wpi::StringRef type) {
+  m_type = type;
+  m_metadataDirty = true;
+}
+
+void ShuffleboardComponentBase::BuildMetadata(
+    std::shared_ptr<nt::NetworkTable> metaTable) {
+  if (!m_metadataDirty) {
+    return;
+  }
+  // Component type
+  if (GetType() == "") {
+    metaTable->GetEntry("PreferredComponent").Delete();
+  } else {
+    metaTable->GetEntry("PreferredComponent").ForceSetString(GetType());
+  }
+
+  // Tile size
+  if (m_width <= 0 || m_height <= 0) {
+    metaTable->GetEntry("Size").Delete();
+  } else {
+    double temp[] = {static_cast<double>(m_width),
+                     static_cast<double>(m_height)};
+    metaTable->GetEntry("Size").SetDoubleArray(temp);
```

Comment:
You can't just pass in the `{...}` directly (should use the `std::initializer_list` overload of `ArrayRef`)?